In [1]:
import numpy as np
import pandas as pd

In [2]:
file_path = r"/content/all_data.csv"

In [3]:
data = pd.read_csv(file_path)

<ipython-input-3-2861b702887f>:1: DtypeWarning: Columns (8,9,31,32,36) have mixed types. Specify dtype option on import or set low_memory=False.
  data = pd.read_csv(file_path)


In [4]:
data.head()

Unnamed: 0  subject  block phase                condition  trial  \
0        37.0        2      1  test  Associative recognition      1   
1        38.0        2      1  test  Associative recognition      2   
2        39.0        2      1  test  Associative recognition      3   
3        40.0        2      1  test  Associative recognition      4   
4        41.0        2      1  test  Associative recognition      5   

   stim.num.left  stim.num.right stim.string.left stim.string.right  ...  \
0            141             722            CLAIM         SCATTERED  ...   
1            809             748          SHARPLY            STRESS  ...   
2            894             352           FORMER            VOLUME  ...   
3            683             189         CONTROLS         REMAINING  ...   
4            857              30            TRACK        APPARENTLY  ...   

   study.partner.left.num  study.partner.right.num  study.partner.left.string  \
0                   722.0                    141.0                  SCATTERED   
1                   747.0                    620.0                     SHARED   
2                   352.0                    894.0                     FORMER   
3                   919.0                    737.0                    WRAPPED   
4                   665.0                    122.0                     READER   

   study.partner.right.string  resp.num  study.partner.resp.num  \
0                       CLAIM       NaN                     NaN   
1                      POURED       NaN                     NaN   
2                      VOLUME       NaN                     NaN   
3                   SENTENCES       NaN                     NaN   
4                        CASH       NaN                     NaN   

   study.partner.resp.string  recall.type  recall.list  \
0                        NaN          NaN          NaN   
1                        NaN          NaN          NaN   
2                        NaN          NaN          NaN   
3                        NaN          NaN          NaN   
4                        NaN          NaN          NaN   

             condition.block  
0  Associative recognition 1  
1  Associative recognition 1  
2  Associative recognition 1  
3  Associative recognition 1  
4  Associative recognition 1  

[5 rows x 39 columns]

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168107 entries, 0 to 168106
Data columns (total 39 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   Unnamed: 0                  168107 non-null  float64
 1   subject                     168107 non-null  int64  
 2   block                       168107 non-null  int64  
 3   phase                       168107 non-null  object 
 4   condition                   168107 non-null  object 
 5   trial                       168107 non-null  int64  
 6   stim.num.left               168107 non-null  int64  
 7   stim.num.right              168107 non-null  int64  
 8   stim.string.left            107886 non-null  object 
 9   stim.string.right           26842 non-null   object 
 10  stim.distractor             168107 non-null  int64  
 11  studied                     168107 non-null  int64  
 12  freq.left                   168107 non-null  int64  
 13  cv.left       

In [6]:
data.describe()

Unnamed: 0        subject          block          trial  \
count  1.681070e+05  168107.000000  168107.000000  168107.000000   
mean   1.692305e+05     231.327500       7.816962      14.984272   
std    1.829088e+05     132.952894       4.274027      11.149121   
min    3.700000e+01       1.000000       1.000000       1.000000   
25%    7.895150e+04     116.000000       4.000000       7.000000   
50%    1.577311e+05     231.000000       8.000000      13.000000   
75%    2.367365e+05     346.000000      11.000000      19.000000   
max    3.129481e+06     462.000000      15.000000     125.000000   

       stim.num.left  stim.num.right  stim.distractor        studied  \
count  168107.000000   168107.000000    168107.000000  168107.000000   
mean      296.427852       74.139328         0.003093       0.709334   
std       307.258274      200.633955         0.147117       0.454071   
min         0.000000        0.000000         0.000000       0.000000   
25%         0.000000        0.000000         0.000000       0.000000   
50%       206.000000        0.000000         0.000000       1.000000   
75%       563.000000        0.000000         0.000000       1.000000   
max       924.000000      924.000000         9.000000       1.000000   

           freq.left        cv.left  ...           resp  study.pos.left  \
count  168107.000000  168107.000000  ...  168107.000000   168107.000000   
mean      282.223090     208.956308  ...       0.375796        9.107836   
std       273.578116     191.063702  ...       0.484329       13.661988   
min         0.000000       0.000000  ...       0.000000        0.000000   
25%         0.000000       0.000000  ...       0.000000        0.000000   
50%       349.000000     292.000000  ...       0.000000        1.000000   
75%       494.000000     375.000000  ...       1.000000       14.000000   
max      1358.000000     500.000000  ...       1.000000       50.000000   

       study.pos.right        kf.left       kf.right  study.partner.left.num  \
count    168107.000000  168107.000000  168107.000000            67322.000000   
mean          4.869958      30.767969       7.614442              464.093833   
std          11.407260      32.497632      20.803612              266.765438   
min           0.000000       0.000000       0.000000                1.000000   
25%           0.000000       0.000000       0.000000              233.000000   
50%           0.000000      27.000000       0.000000              465.000000   
75%           0.000000      50.000000       0.000000              696.000000   
max          40.000000     197.000000     197.000000              924.000000   

       study.partner.right.num      resp.num  study.partner.resp.num  \
count             26842.000000  63721.000000            60115.000000   
mean                463.708889    460.347845              460.821658   
std                 265.750905    267.248810              267.419900   
min                   1.000000      1.000000                1.000000   
25%                 235.000000    229.000000              229.000000   
50%                 462.000000    458.000000              459.000000   
75%                 694.000000    693.000000              693.000000   
max                 924.000000    924.000000              924.000000   

        recall.list  
count  20339.000000  
mean       4.933969  
std        4.636837  
min        0.000000  
25%        1.000000  
50%        4.000000  
75%        9.000000  
max       15.000000  

[8 rows x 26 columns]

In [7]:
data.isnull().sum()

Unnamed: 0                         0
subject                            0
block                              0
phase                              0
condition                          0
trial                              0
stim.num.left                      0
stim.num.right                     0
stim.string.left               60221
stim.string.right             141265
stim.distractor                    0
studied                            0
freq.left                          0
cv.left                            0
old.left                           0
freq.right                         0
cv.right                           0
old.right                          0
rt                             50157
distractor.resp                    0
resp                               0
resp.string                    97306
resp.type                          0
resp.type.rescore              82478
study.pos.left                     0
study.pos.right                    0
kf.left                            0
kf.right                           0
resp.string.corr               93880
study.partner.left.num        100785
study.partner.right.num       141265
study.partner.left.string     100785
study.partner.right.string    141265
resp.num                      104386
study.partner.resp.num        107992
study.partner.resp.string     107992
recall.type                   147768
recall.list                   147768
condition.block                    0
dtype: int64

In [9]:
data['phase'].unique()

array(['test'], dtype=object)

In [10]:
data['condition'].unique()

array(['Associative recognition', 'Free recall', 'Cued recall',
       'Lexical decision', 'Single recognition'], dtype=object)

In [12]:
print(data['phase'].value_counts())

phase
test    168107
Name: count, dtype: int64


In [13]:
data['condition'].value_counts()

condition
Free recall                60221
Lexical decision           27121
Cued recall                27035
Single recognition         26888
Associative recognition    26842
Name: count, dtype: int64

### condition column has what we are looking for - memory task

I will focus on this column

- Free recall
- Lexical decision
- Cued recall
- Single-item recognition (mapped as "Single recognition")
- Associative recognition

In [14]:
tasks = ['Free recall', 'Lexical decision', 'Cued recall', 'Single recognition', 'Associative recognition']


split data acc to diff tasks

In [15]:
task_data = {task: data[data['condition'] == task] for task in tasks}

In [19]:
for task, df in task_data.items():
    print(f"\n--- {task} Task ---")
    print(f"Rows: {df.shape[0]}")
    print(f"Missing values:\n{df.isnull().sum()}")


--- Free recall Task ---
Rows: 60221
Missing values:
Unnamed: 0                        0
subject                           0
block                             0
phase                             0
condition                         0
trial                             0
stim.num.left                     0
stim.num.right                    0
stim.string.left              60221
stim.string.right             60221
stim.distractor                   0
studied                           0
freq.left                         0
cv.left                           0
old.left                          0
freq.right                        0
cv.right                          0
old.right                         0
rt                            50157
distractor.resp                   0
resp                              0
resp.string                     292
resp.type                         0
resp.type.rescore              1626
study.pos.left                    0
study.pos.right                   0
kf.left   

I should focus on identifying relevant columns regarding each task.

In [21]:
cleaned_tasks = {}

1. --- Free recall Task ---
Rows: 60221
- Focus: resp.string, trial, subject


In [26]:
cleaned_tasks['Free recall'] = task_data['Free recall'][['subject', 'trial', 'resp.string']].dropna()
cleaned_tasks['Free recall']['resp.string'].unique()

array(['senate', 'fiexd', 'fixed', ..., 'MINDS', 'ENORMOUS', 'ROUTE'],
      dtype=object)

In [28]:
cleaned_tasks['Free recall']['trial'].unique()

array([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
        14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,
        27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,
        40,  41,  43,  44,  45,  46,  48,  49,  50,  51,  52,  53,  54,
        55,  56,  57,  58,  59,  60,  61,  62,  63,  64,  65,  66,  67,
        68,  69,  70,  71,  72,  73,  74,  75,  76,  77,  78,  79,  80,
        81,  83,  84,  85,  42,  47,  86,  87,  88,  89,  90,  91,  92,
        93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103, 104, 105,
       106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118,
       119, 120, 121, 122, 123, 124, 125,  82])

In [29]:
cleaned_tasks['Free recall']['subject'].unique()

array([  2,   3,   4,   5,   6,   7,   8,   9,  14,  15,  16,  17,  18,
        19,  20,  22,  24,  25,  26,  27,  28,  29,  30,  31,  33,  34,
        35,  36,  37,  38,  39,  40,  41,  42,  44,  45,  46,  47,  48,
        50,  51,  52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,
        63,  65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,
        77,  78,  79,  80,  81,  84,  86,  87,  88,  89,  90,  91,  92,
        94,  95,  96,  97,  98,  99, 100, 101, 102, 103, 104, 106, 107,
       108, 109, 110, 111, 113, 114, 115, 117, 119, 120, 121, 122, 125,
       128, 129, 130, 132, 135, 136, 137, 138, 139, 141, 142, 143, 144,
       145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157,
       158, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 171, 172,
       173, 174, 175, 176, 177, 178, 180, 181, 182, 183, 184, 185, 186,
       187, 188, 189, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201,
       202, 203, 204, 206, 207, 208, 209, 210, 211, 212, 213, 21

2. Lexical Decision - classify the stimulus
- subject - identification for participation
- stim.string.left - word to do classification on
- resp - decision
- rt - response time

In [31]:
cleaned_tasks['Lexical decision'] = task_data['Lexical decision'][['subject', 'stim.string.left', 'resp', 'rt']].dropna()
cleaned_tasks['Lexical decision']['stim.string.left'].unique()

array(['APARTMENT', 'RASSING', 'CONWRESHLY', ..., 'ORGANISM', 'COLUMN',
       'COAT'], dtype=object)

In [32]:
cleaned_tasks['Lexical decision']['resp'].unique()

array([1, 0])

In [34]:
cleaned_tasks['Lexical decision']['rt'].unique()[:5]

array([0.94, 1.77, 2.51, 0.58, 1.57])

I dont think response time will be useful for llm work

3. Cued Recall - gien one word recall the pair word

- subject - id
- trial
- stim.string.left - cue word
- resp.string - participant response

In [36]:
cleaned_tasks['Cued recall'] = task_data['Cued recall'][['subject', 'trial', 'stim.string.left', 'resp.string']].dropna()
cleaned_tasks['Cued recall']['stim.string.left'].unique()[:20]

array(['SERIOUSLY', 'PARTIES', 'REACHES', 'SAVED', 'CONNECTED', 'MISTAKE',
       'SISTERS', 'MUD', 'TOPIC', 'SCREEN', 'BAG', 'INTERNAL', 'ACID',
       'LIBRARY', 'WINDS', 'EXPENSIVE', 'ROPE', 'AGES', 'GUIDE', 'BITS'],
      dtype=object)

In [37]:
cleaned_tasks['Cued recall']['resp.string'].unique()[:20]

array(['tough', 'bedroom', 'continue', 'location', 'recieved', 'latin',
       'door', 'choice', 'motor', 'man', 'outer', 'rain', 'support',
       'golden', 'writers', 'tissue', 'consent', 'degree',
       'relationships', 'procedure'], dtype=object)

4. Single Recognition - check if a single word is part of a studied list

- subject - id
- stim.string.left - recognition word
- resp - decision

In [38]:
cleaned_tasks['Single recognition'] = task_data['Single recognition'][['subject', 'stim.string.left', 'resp']].dropna()

In [40]:
cleaned_tasks['Single recognition']['stim.string.left'].unique()[:20]

array(['PROVE', 'MARKS', 'OUTPUT', 'FILE', 'PRINT', 'SIZES', 'TUBE',
       'FEARED', 'POLLUTION', 'DESIGN', 'OPPORTUNITIES', 'ESCAPE',
       'STRING', 'STANDS', 'HAT', 'SLIPPED', 'OCCUPIED', 'HARMFUL',
       'SPLIT', 'SPOKEN'], dtype=object)

In [41]:
cleaned_tasks['Single recognition']['resp'].unique()[:20]

array([0, 1])

5. Associative Recognition - check if pair of words shown in study phase

- subject - id
- stim.string.left - 1st word of pair
- stim.string.right - 2nd word of pair
- resp - decision

In [42]:
cleaned_tasks['Associative recognition'] = task_data['Associative recognition'][['subject', 'stim.string.left', 'stim.string.right', 'resp']].dropna()


In [43]:
cleaned_tasks['Associative recognition']['stim.string.left'].unique()[:20]

array(['CLAIM', 'SHARPLY', 'FORMER', 'CONTROLS', 'TRACK', 'PROFIT',
       'LEATHER', 'INSTRUCTIONS', 'WINGS', 'SENTENCES', 'SHARED', 'FIRMS',
       'EFFICIENT', 'CAMP', 'BLACKS', 'MARRIAGE', 'EXTERNAL', 'PROPOSED',
       'NEWSPAPERS', 'SHINING'], dtype=object)

In [44]:
cleaned_tasks['Associative recognition']['stim.string.right'].unique()[:20]

array(['SCATTERED', 'STRESS', 'VOLUME', 'REMAINING', 'APPARENTLY',
       'TIGHT', 'POURED', 'READER', 'DESIRE', 'RESULTING', 'CASH',
       'AGENCY', 'WRAPPED', 'CHARACTERISTIC', 'PULLING', 'RUBBER',
       'PUBLISHED', 'WARS', 'ATTEMPTS', 'MAGIC'], dtype=object)

In [45]:
cleaned_tasks['Associative recognition']['resp'].unique()[:20]

array([1, 0])

In [46]:
for task, df in cleaned_tasks.items():
    print(f"{task}: {df.shape}")

Free recall: (59929, 3)
Lexical decision: (27121, 4)
Cued recall: (10872, 4)
Single recognition: (26888, 3)
Associative recognition: (26842, 4)


### I GOT THIS MEMORY DETAILS AND PROCEDURE FROM - `Design and Procedure Page no 14 onward` 